### ISSUE TITLE
Prometheus can't scrape mTLS pods in v1.5

### ISSUE THREAD

**Issue by hagen93** [istio/istio#22391]
**Bug description**
There seems to be an issue where the Prometheus shipped with Istio v1.5 can no longer scrape pods with the "prometheus.io/scrape: true" annotation. This seems to be because the job "kubernetes-pods-istio-secure" is gated behind the flag [".Values.security.enabled"](https://github.com/istio/istio/blob/master/manifests/istio-telemetry/prometheus/templates/configmap.yaml#L253) which seems to not be present in the default profile when installing Istio with istioctl. Adding the flag manually by calling istioctl with the parameter "--set values.security.enabled=true" seems to resolve the issue. After installing Istio as described the workaround from #21843 seems to also be needed for the "kubernetes-pods-istio-secure" job to get the appropriate certificates.

**Expected behavior**
Prometheus should be able to scrape pods protected with mTLS.

**Steps to reproduce the bug**
- Install istio with "istioctl manifest apply"
- Deploy a pod with the annotation "prometheus.io/scrape: true"
- Check Prometheus targets. The pod should not have been picked up as the "kubernetes-pods-istio-secure" job is missing and the "kubernetes-pods" drops mTLS pods because of the annotation "sidecar.istio.io/status".

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
$ istioctl version --remote
client version: 1.5.0
control plane version: 1.5.0
data plane version: 1.5.0 (17 proxies)
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-12T13:43:46Z", GoVersion:"go1.13.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:07:13Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}

**How was Istio installed?**
With istioctl CLI

**Environment where bug was observed (cloud vendor, OS, etc)**
Ubuntu 18.04 LTS running Microk8s (not the bundled Istio)

**Comment by hagen93** [comment]
As far as I can tell this broke in 2d8249df15868ab1bdd116d5368285613a4fd623 when ".Values.prometheus.security.enabled" was changed to ".Values.security.enabled". The default profile still has [".Values.prometheus.security.enabled=true"](https://github.com/istio/istio/blob/master/operator/data/profiles/default.yaml#L551) so just reverting the change seems to be enough to solve the issue, but I don't know what other parts that might affect. If the change from 2d8249df15868ab1bdd116d5368285613a4fd623 is neccecary then inserting an "enabled: true" [here](https://github.com/istio/istio/blob/master/operator/data/profiles/default.yaml#L482) should also sufficient to fix the problem as far as I can tell. I can submit a PR to fix the problem if someone could tell me which one of the solutions are appropriate. :smile:

**Comment by howardjohn** [comment]
It probably should be behind `.Values.prometheus.provisionPrometheusCert` @lei-tang

**Comment by sibleyd** [comment]
I attempted setting values.security.enabled to true as suggested by @hagen93, at which point the kubernetes-pods-istio-secure scrape job is added to the configmap, however prometheus can't actually create the scrape pool because the deployment still looks for the istio.default secret to get the certs.

**Comment by hagen93** [comment]
@sibleyd I believe the certificate problem is covered by #21843. :smile:

**Comment by sibleyd** [comment]
Yep, that does resolve that issue, thanks. With that I have the kubernetes-pods-istio-secure job up and hitting my pod.

**Comment by hagen93** [comment]
@howardjohn @lei-tang Do you guys know if `.Values.prometheus.provisionPrometheusCert` is correct yet? :)

**Comment by lei-tang** [comment]
As the default profile still has ".Values.prometheus.security.enabled=true" and in the original implementation, the job "kubernetes-pods-istio-secure" is gated behind the flag ".Values.prometheus.security.enabled", in my opinion, we may want to go back to the original implementation (i.e.,  the job "kubernetes-pods-istio-secure" is gated behind the flag ".Values.prometheus.security.enabled").

**Comment by bethmage** [comment]
This is still not fixed in istio v1.5.4

**Comment by lei-tang** [comment]
In Istio 1.5.4, the job "kubernetes-pods-istio-secure" is still gated behind the flag ".Values.security.enabled" whereas the default profile has ".Values.prometheus.security.enabled=true". Thus, the job "kubernetes-pods-istio-secure" is not started with the default profile.

@hagen93 Thank you for offering to provide a PR to fix this problem! Gating the job "kubernetes-pods-istio-secure" behind ".Values.prometheus.security.enabled=true" (which is the original implementation) sounds good to me.

@bethmage Before the PR from @hagen93 is merged, you may try the workarounds mentioned in this issue, such as configuring ".Values.security.enabled" as true in your installation profile.

**Comment by bethmage** [comment]
Thank you. I did so.

**Comment by hagen93** [comment]
@lei-tang Looks like someone already fixed it. The [configmap](https://github.com/istio/istio/blob/dad709d38bede911ed4f7732a64d9444c78e11dc/manifests/charts/istio-telemetry/prometheus/templates/configmap.yaml#L257) now gates it behind `and .Values.prometheus.provisionPrometheusCert (not .Values.meshConfig.enablePrometheusMerge)` which seems to be the values provided by the default profile [here](https://github.com/istio/istio/blob/dad709d38bede911ed4f7732a64d9444c78e11dc/manifests/profiles/default.yaml#L477) and [here](https://github.com/istio/istio/blob/dad709d38bede911ed4f7732a64d9444c78e11dc/manifests/profiles/default.yaml#L11).

**Comment by lei-tang** [comment]
@hagen93 Thanks! The fix you mentioned is in the master branch. Can you cherrypick the change related to the job "kubernetes-pods-istio-secure" to the release-1.5 branch?

**Comment by hagen93** [comment]
@lei-tang Sure, but the fix is part of a much larger commit, f44354b25790a88f6581227d22f00948b51a7d45. We should probably just cherry-pick the relevant changes right? Otherwise I'm afraid we'll break other stuff :)

**Comment by lei-tang** [comment]
@hagen93 That is right, just cherry-pick the relevant changes/lines :)

**Comment by hagen93** [comment]
@lei-tang Cool, I'll have an MR in soon :)

**Comment by hagen93** [comment]
@lei-tang I think the MR I just linked should fix the problem, but I am a bit unsure of how to test the changes. Tried building istioctl with go build, but the changes seems to not be included in the final binary. I suspect that is because go ends up pulling stuff from the GitHub repo instead of using my local copy. Had a quick look at the Makefile to see if there was a make profile that would build it for me, but it is a bit complex. Do you know of an easy way to test the changes?

**Comment by lei-tang** [comment]
@hagen93 Regarding to test the changes, "export HUB=YOUR-HUB; export TAG=YOUR-TAG; make push" should build istio (including istioctl) and push the binaries to the HUB you specified. After "make push", you can use the istioctl built locally along with other istio binaries pushed to the HUB to test the changes you made.

**Comment by hagen93** [comment]
@lei-tang I think I fixed it, but to compile the fix into the binary it looks like I need to regenerate operator/pkg/vfs/assets.gen.go, do you know which make command triggers the regeneration? :)

**Comment by lei-tang** [comment]
@hagen93 IIRC, the command to trigger the regeneration is "make gen".

**Comment by hagen93** [comment]
@lei-tang I ended up only gating the job behind `.Values.prometheus.provisionPrometheusCert` as `.Values.meshConfig.enablePrometheusMerge` seems to be intertwined with other things from f44354b25790a88f6581227d22f00948b51a7d45.

`make gen` edited a lot of files in addition to operator/pkg/vfs/assets.gen.go but I only included the one file in the MR.

Otherwise AFAIK the MR should be good now, pending the CI tests. :)
